File: musicShopProject/tentashop/src/domain/orders/orderBLank.ts
import { OrderState } from "./enum/orderState";
import { Order } from "./order";
import { OrderItemBlank } from "./orderItemBLank";

export interface OrderBlank{
    id: string | null,
    price: number | null,
    clientPhoneNumber : string | null,
    ClientId : string | null,
    AddressId : string | null,
    CompletedDateTimeUtc: Date | null,
    CreatedDateTimeUtc: Date | null,
    State: OrderState | null,
    OrderNumber: number | null,
    OrderItems: OrderItemBlank[] | []
}

export namespace OrderBlank {
    export function getEmpty(): OrderBlank {
        return {
                id:  null,
                price:  null,
                clientPhoneNumber : null,
                ClientId :  null,
                AddressId : null,
                CompletedDateTimeUtc: null,
                CreatedDateTimeUtc:  null,
                State: null,
                OrderNumber: null,
                OrderItems:  []
        };
    }

    export function toProductBlank(order: Order): OrderBlank {
        return {
            id: order.id,
            price:  order.price,
            clientPhoneNumber : order.clientPhoneNumber,
            ClientId :  order.client.id,
            AddressId : order.address.id,
            CompletedDateTimeUtc: order.completedDateTimeUtc,
            CreatedDateTimeUtc:  order.createdDateTimeUtc,
            State: order.state,
            OrderNumber: order.orderNumber,
            OrderItems:  []
        };
    }
 }

File: musicShopProject/Model/Orders/OrderBlank.cs
using musicShopProject.Model.Orders.enums;

namespace musicShopProject.Model.Orders;

public partial class OrderBlank
{
    public Guid? Id { get; set; }
    public Decimal? Price { get; set; }
    public String? ClientPhoneNumber { get; set; }
    public Guid? ClientId { get; set; }
    public Guid? AddressId { get; set; }
    public DateTime? CompletedDateTimeUtc { get; set; }
    public DateTime? CreatedDateTimeUtc { get; set; }
    public OrderState? State { get; set; }
    public Int32? OrderNumber { get; set; }
    public OrderItemBlank[] OrderItems { get; set; }
}

public partial class OrderBlank
{
    public class Validated
    {
        public Guid? Id { get; }
        public Decimal? Price { get; }
        public String? ClientPhoneNumber { get; }
        public Guid? ClientId { get; }
        public Guid? AddressId { get; }
        public DateTime? CompletedDateTimeUtc { get; }
        public DateTime? CreatedDateTimeUtc { get; }
        public OrderState? State { get; }
        public Int32? OrderNumber { get; }
        public OrderItemBlank.Validated[] ValidatedOrderItemBlanks { get; }

        public Validated(Guid id, Decimal price, String clientPhoneNumber, Guid clientId, Guid addresId,
            DateTime completedDateTimeUtc, DateTime createdDateTimeUtc, OrderState state, Int32 orderNumber, OrderItemBlank.Validated[] validatedOrderItemBlanks)
        {
            Id = id;
            Price = price;
            ClientPhoneNumber = clientPhoneNumber;
            ClientId = clientId;
            AddressId = addresId;
            CompletedDateTimeUtc = completedDateTimeUtc;
            CreatedDateTimeUtc = createdDateTimeUtc;
            State = state;
            OrderNumber = orderNumber;
            ValidatedOrderItemBlanks = validatedOrderItemBlanks;
        }
    }
}
